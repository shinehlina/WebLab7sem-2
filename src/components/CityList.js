import React from "react";
import City from "./City";
import { Preloader } from "react-materialize";
import Row from "react-materialize/lib/Row";
import Col from "react-materialize/lib/Col";

class CityList extends React.Component {
  onBtnClick = e => {
    if (e.keyCode === 13) {
      const city = e.target.value;
      if (
        !this.props.favoriteCities
          .map(c => c.cityName.toUpperCase())
          .includes(city.toUpperCase())
      ) {
        this.props.addCity(city);
      }
      e.target.value = "";
      e.preventDefault();
    }
  };

  createList(props) {
    if (props.favoriteCities.length === 0) {
      return <div>No favorite cities</div>;
    }
    let cityNameList = [];
    for (let i = 0; i < props.favoriteCities.length; i++) {
      let cityInfo = props.favoriteCities[i];
      if (cityInfo !== null) {
        cityNameList.push(
          <li key={cityInfo.cityName}>
            <City
              index={i}
              cityName={cityInfo.cityName}
              longtitude={cityInfo.data.coord.lon}
              latitude={cityInfo.data.coord.lat}
              description={cityInfo.data.weather[0].description}
              windSpeed={cityInfo.data.wind.speed}
              temperature={cityInfo.data.main.temp}
              humidity={cityInfo.data.main.humidity}
              pressure={cityInfo.data.main.pressure}
              deleteCity={props.deleteCity}
            />
          </li>
        );
      }
    }
    return cityNameList;
  }

  render() {
    return (
      <div>
        <Row>
          <Col className="valign-wrapper" s={6}>
            <h5 className="">Favorites</h5>
          </Col>
          <Col className="valign-wrapper" s={6}>
            <form>
              <input
                id="city"
                type="text"
                size="40"
                placeholder="Enter your city"
                onKeyDown={this.onBtnClick}
              />
            </form>
          </Col>
        </Row>
        {this.props.isFetching ? (
          <Preloader size="big" />
        ) : (
          <ul>{this.createList(this.props)}</ul>
        )}
      </div>
    );
  }
}

export default CityList;
