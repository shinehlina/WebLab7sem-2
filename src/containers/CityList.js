import React from "react";

class CityList extends React.Component {
  //   rendenCity(name, temperature) {
  //     return <City cityName={name} temperature={temperature} />;
  //   }

  onBtnClick = e => {
    if (e.keyCode === 13) {
      const city = e.target.value;
      this.props.addCity(city);
      e.target.value = "";
      e.preventDefault();
    }
  };

  createList(props) {
    let cityNameList = [];
    for (let i = 0; i < props.favoriteCities.length; i++) {
      let cityName = props.favoriteCities[i];
      if (cityName !== null) {
        // list.push(this.rendenCity(city.name, city.temperature));
        cityNameList.push(<div key={i}>{cityName}</div>);
      }
    }
    if (cityNameList.length === 0) {
      cityNameList.push(<div>No favorite cities</div>);
    }
    return cityNameList;
  }

  render() {
    return (
      <div>
        <h1> Города </h1>
        <form>
          <input
            id="city"
            type="text"
            size="40"
            placeholder="Enter your city"
            onKeyDown={this.onBtnClick}
          />
        </form>
        {this.createList(this.props)}
      </div>
    );
  }
}

export default CityList;
