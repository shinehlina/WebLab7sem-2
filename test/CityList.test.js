import React from "react";
import { shallow } from "./enzyme";
import { shallowToJson } from "enzyme-to-json";
import CityList from "../src/components/CityList";

export let favoriteCities = [
  {
    cityName: "Kiev",
    data: {
      coord: {
        lon: 30.33,
        lat: 60.03
      },
      weather: [
        {
          main: "Rain",
          description: "light rain",
          icon: "1"
        }
      ],
      main: {
        temp: 278.02,
        pressure: 1028,
        humidity: 93
      },
      wind: {
        speed: 3
      }
    }
  }
];

it("CityList success didn't changed", () => {
  const app = shallow(
    <CityList
      favoriteCities={favoriteCities}
      areFetching={false}
      error={""}
      addCity={() => undefined}
      deleteCity={() => undefined}
      getCityList={() => undefined}
    />
  );
  expect(shallowToJson(app)).toMatchSnapshot();
});

it("CityList error no cities didn't changed", () => {
  const app = shallow(
    <CityList
      areFetching={false}
      favoriteCities={[]}
      error={"Error"}
      addCity={() => undefined}
      deleteCity={() => undefined}
      getCityList={() => undefined}
    />
  );
  expect(shallowToJson(app)).toMatchSnapshot();
});

it("CityList error has cities didn't changed", () => {
  const app = shallow(
    <CityList
      areFetching={false}
      favoriteCities={favoriteCities}
      error={"Error"}
      addCity={() => undefined}
      deleteCity={() => undefined}
      getCityList={() => undefined}
    />
  );
  expect(shallowToJson(app)).toMatchSnapshot();
});

it("CityList loading didn't changed", () => {
  const app = shallow(
    <CityList
      areFetching={true}
      error={""}
      addCity={() => undefined}
      deleteCity={() => undefined}
      getCityList={() => undefined}
    />
  );
  expect(shallowToJson(app)).toMatchSnapshot();
});
