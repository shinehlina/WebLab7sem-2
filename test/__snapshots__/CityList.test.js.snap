// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList error has cities didn't changed 1`] = `
<div>
  <Modal
    actions={
      Array [
        <Button
          flat={true}
          modal="close"
          node="button"
          waves="green"
        >
          Close
        </Button>,
      ]
    }
    bottomSheet={false}
    fixedFooter={false}
    header="Error"
    id="modal-0"
    open={true}
    options={
      Object {
        "dismissible": true,
        "endingTop": "10%",
        "inDuration": 250,
        "onCloseEnd": null,
        "onCloseStart": null,
        "onOpenEnd": null,
        "onOpenStart": null,
        "opacity": 0.5,
        "outDuration": 250,
        "preventScrolling": true,
        "startingTop": "4%",
      }
    }
  >
    Error
     
  </Modal>
</div>
`;

exports[`CityList error no cities didn't changed 1`] = `
<div>
  <Modal
    actions={
      Array [
        <Button
          flat={true}
          modal="close"
          node="button"
          waves="green"
        >
          Close
        </Button>,
      ]
    }
    bottomSheet={false}
    fixedFooter={false}
    header="Error"
    id="modal-0"
    open={true}
    options={
      Object {
        "dismissible": true,
        "endingTop": "10%",
        "inDuration": 250,
        "onCloseEnd": null,
        "onCloseStart": null,
        "onOpenEnd": null,
        "onOpenStart": null,
        "opacity": 0.5,
        "outDuration": 250,
        "preventScrolling": true,
        "startingTop": "4%",
      }
    }
  >
    Error
     
  </Modal>
</div>
`;

exports[`CityList loading didn't changed 1`] = `
<div>
  <Row>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <h5
        className=""
      >
        Favorites
      </h5>
    </Col>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <form>
        <input
          id="city"
          onKeyDown={[Function]}
          placeholder="Enter your city"
          size="40"
          type="text"
        />
      </form>
    </Col>
  </Row>
  <div
    className="center-align"
  >
    <Preloader
      active={true}
      color="blue"
      flashing={false}
      size="big"
    />
  </div>
</div>
`;

exports[`CityList success didn't changed 1`] = `
<div>
  <Row>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <h5
        className=""
      >
        Favorites
      </h5>
    </Col>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <form>
        <input
          id="city"
          onKeyDown={[Function]}
          placeholder="Enter your city"
          size="40"
          type="text"
        />
      </form>
    </Col>
  </Row>
  <Row>
    <Col
      key="Kiev"
      m={6}
      s={12}
    >
      <City
        cityName="Kiev"
        deleteCity={[Function]}
        index={0}
      />
    </Col>
  </Row>
</div>
`;
